### DISCUSSION TITLE
Add More Logging?

### DISCUSSION THREAD

**Discussion by r-pascua** [HERA-Team/hera_sim#128] [Ideas]
In #111, I suggested that we may want to add more logging statements to various simulation tasks in the repo. I think it may prove useful for steps that are relatively (with respect to other simulation tools within this repo) compute-intensive or things that do not scale nicely. Rather than plan to address the topic of this issue, I would like to open some discussion about it--whether we think it's worth implementing, possible use cases, things we would like to track, etc. Input from anyone who interacts with `hera_sim` at some level is greatly appreciated.

**Comment by steven-murray** [comment]
I think this is a great idea, and can be implemented without causing much
user difference if logging to INFO level or lower.

Good for tracking long computations, or debugging certain parts if we know
they're tricky.

**Comment by piyanatk** [comment]
I second this. 

What is the proper way to implement this? Via the `logging` module?

**Reply by steven-murray** [reply]
Absolutely. Use a hierarchy of loggers. For the CLI, I definitely recommend using the `rich` library, which can add a `RichHandler` handler that makes the log output really nice to look at.
